py3status/modules/weather_owm.py

```diff
@@ -0,0 +1,739 @@
+```
+weather_owm {
+  api_key = '...'
+
+  icons {
+    i200 = "☔"
+    i230_i232 = "🌧"
+
+    clouds = "☁"
+  }
+
+  temp_color = True
+}
+```
+Outputs: 🌫: ○: 59° foggy, ⛅ ☼ 🌧`
+- Currently foggy, 59° F outside, with forecast of cloudy tomorrow, sunny the
+  next day, then rainy
+
+
+@author alexoneill
+@licence MIT
+'''
+
+import datetime
+
+
+# API information
+OWM_API = '2.5'
+OWM_CURR_ENDPOINT = 'http://api.openweathermap.org/data/%s/weather?' \
+    'APPID=%s&lat=%f&lon=%f&lang=%s'
+OWM_FUTURE_ENDPOINT = 'http://api.openweathermap.org/data/%s/forecast?' \
+    'APPID=%s&lat=%f&lon=%f&lang=%s&cnt=%%d'
+IP_ENDPOINT = 'http://ip-api.com/json'
+
+# Paths of information to extract from JSON
+IP_LAT = '//lat'
+IP_LNG = '//lon'
+OWM_WEATHER_ICON = '//weather:0/id'
+OWM_CLOUD_COVER = '//clouds/all'
+OWM_RAIN = '//rain/3h'
+OWM_SNOW = '//snow/3h'
+OWM_WIND = '//wind'
+OWM_HUMIDITY = '//main'
+OWM_PRESSURE = '//main'
+OWM_TEMP = '//main'
+OWM_SUNRISE = '//sys/sunrise'
+OWM_SUNSET = '//sys/sunset'
+OWM_DESC = '//weather:0/main'
+OWM_DESC_LONG = '//weather:0/description'
+
+# Units constants
+RAIN_UNITS = set(['cm', 'ft', 'in', 'mm', 'm', 'yrd'])
+SNOW_UNITS = RAIN_UNITS
+TEMP_UNITS = set(['c', 'f', 'k'])
+WIND_UNITS = set(['fsec', 'msec', 'mph', 'kmh'])
+
+# Conversion factors
+IN_FROM_MM = 0.0393701
+FT_FROM_METER = 3.28084
+KMH_FROM_MSEC = 0.277778
+MPH_FROM_MSEC = 2.23694
+
+# Thresholds defaults
+THRESHOLDS = [
+    (-100, '#0FF'),
+    (0, '#00F'),
+    (50, '#0F0'),
+    (150, '#FF0')
+]
+
+
+class OWMException(Exception):
+    pass
+
+
+class Py3status:
+
+    api_key = None
+    cache_timeout = 600
+    forecast_days = 3
+    forecast_include_today = False
+    forecast_text_separator = ' '
+    format = '{icon}: {temperature} {desc}, {forecast}'
+    format_clouds = '{icon}: {coverage}%'
+    format_forecast = '{icon}'
+    format_humidity = '{icon}: {humidity}%'
+    format_pressure = '{icon}: {pressure} hPa'
+    format_rain = '{icon}: {amount} inches'
+    format_snow = '{icon}: {amount} inches'
+    format_sunrise = '{icon}: %X'
+    format_sunset = '{icon}: %X'
+    format_temperature = u'{icon}: {current}°'
+    format_wind = '{icon}: {speed} mph'
+    icon_atmosphere = u'🌫'
+    icon_breeze = u'☴'
+    icon_cloud = u'☁'
+    icon_extreme = u'⚠'
+    icon_humidity = u'●'
+    icon_pressure = u'◌'
+    icon_rain = u'🌧'
+    icon_snow = u'❄'
+    icon_sun = u'☼'
+    icon_temperature = u'○'
+    icon_thunderstorm = u'⛈'
+    icons = None
+    lang = 'en'
+    location = None
+    rain_unit = 'in'
+    request_timeout = 10
+    snow_unit = 'in'
+    temp_color = False
+    temp_unit = 'f'
+    thresholds = THRESHOLDS
+    wind_unit = 'mph'
+
+    def _get_icons(self):
+        if self.icons is None:
+            self.icons = {}
+
+        # Defaults for weather ranges
+        defaults = {
+            'i200_i299': self.icon_thunderstorm,
+            'i300_i399': self.icon_rain,
+            'i500_i599': self.icon_rain,
```

Comment:
Yeah, this is correct. These ranges of icons are for rain, and the extreme ones are correct as well. Thanks for checking! :)